### ISSUE TITLE
Scroll is not Fluid

### ISSUE THREAD

**Issue by GoogleCodeExporter** [JustLoy77/ebookdroid#50]
```
Well, is simple scrolling is not working well, it is not fluid, the movement of 
the page when scrolling is "step by step"
This happens with the page already loaded in memory, so it is not that is being 
slow because of page being loaded.
This is most noticeable on djvu documents. In pdf's is less noticeable.
This was not an issue neither in Vudroid plus or Ebookdroid first release the 
scrolling was perfect smooth as hell. The problem started at 1.1.0, to the 
present 1.2.0

What version of the product are you using? On what operating system?
all except first release, Android 2.1

What device are you using?
Flytouch 2. 
A 10 inch china tablet with Android 2.1 and 1024x600 screen resolution

Please provide any additional information below.
First of all i can't believe what you did to this program, is amazing, you 
really are inspired, very good job, very handy and useful options, very good 
ideas congratulations!!

Ps: I have two ideas for future releases

1-"Horizontal scroll lock" (this is an old idea jaja, besides i think that it 
would be very useful and won't hurt anyone if it is there and you don't want to 
use it)

2-An "offset" of page numbers.
I mean, you know that there are some documents that do not coincide the page in 
the real book and the page in the document(like scanned djvu's).
So for example if the page number 1 of the book is actually in page 16 of the 
document it would be handy to "set as first page" page 16.
So then all pages would coincide and you won't need to do maths when you have 
page reference numbers.
Obviously page 1 to 15 would have negatives numbers and zero, but i think that 
won't be a problem.
Maybe this could be added to the bookmarks menus you could add a button that 
says: "set as first page", so you go to the real first page, enter to bookmarks 
and press that button and walaaa a perfect numbered document.

thanks
```

Original issue reported on code.google.com by `lanos...@gmail.com` on 8 Sep 2011 at 5:16

**Comment by GoogleCodeExporter** [comment]
```
Fully agree. Very good and objective issue, except "PS.1 - Horizontal scroll 
lock". This issue has already been discussed, but in Russian:
http://code.google.com/p/ebookdroid/issues/detail?id=1
```

Original comment by `imr...@gmail.com` on 8 Sep 2011 at 9:04

**Comment by GoogleCodeExporter** [comment]
```
>>> Well, is simple scrolling is not working well, it is not fluid, the 
movement of the page when scrolling is "step by step"

What do you mean under "simple scrolling" ? 

Scrolling using touch movement, tap zones, hardware buttons, etc?

PS: please not mix different problems and enhancements in one issue. It is hard 
to manage such ones.

```

Original comment by `Alexander.V.Kasatkin@gmail.com` on 8 Sep 2011 at 9:26
- Changed state: **Discussion**

**Comment by GoogleCodeExporter** [comment]
```
Sorry, i won't mix them again. But i don't know how to set "Enhancement" in 
types, so i added in PS here.
Sorry also, about not being clear enough, but i'm from Argentina, so sometimes 
is difficult to express exactly what i mean.
**********

I mean Touch movement. With both "Dragging" and "Flicking" as explained here in 
Touch Gesture Reference Guide:
http://www.lukew.com/touch/TouchGestureGuide.pdf

Flicking is a fast movement generally, but Dragging is generally slow and the 
page also has not fluid movement.
(By the way this happens with tap zones disabled and within the same, already 
loaded, page.)

On other side, Hardware buttons and tap zones are "step by step" scrolling by 
default and they work fine, when used.
```

Original comment by `lanos...@gmail.com` on 8 Sep 2011 at 5:49

**Comment by GoogleCodeExporter** [comment]
```
Here are made two videos showing the issue, ovbiously the issue is most 
noticeable and annoying in person:

*ebookdroid first release with djvu document, No problems:
http://www.youtube.com/watch?v=ZSDyAXbz2FY

*ebookdroid 1.2.0 with a djvu and a pdf document(in that order)
http://www.youtube.com/watch?v=AQjW-IxUJHU
```

Original comment by `lanos...@gmail.com` on 9 Sep 2011 at 5:52

**Comment by GoogleCodeExporter** [comment]
```
This is an example document where this issue is well noticed (at least for me):
http://www.megaupload.com/?d=NBU46E3W
```

Original comment by `lanos...@gmail.com` on 16 Sep 2011 at 8:29

**Comment by GoogleCodeExporter** [comment]
```
Possibly fixed in 1.3
```

Original comment by `Alexander.V.Kasatkin@gmail.com` on 3 Oct 2011 at 8:15
- Changed state: **Fixed**

**Comment by GoogleCodeExporter** [comment]
```
With 1.2.2-rc2 the scrolling while "dragging" is smoother, close to perfection 
in pdf and a lot better in djvu but a bit "jerky" yet.Good work
But with "fling" i can't tell, because i was enable to "put the page in motion" 
see issue 51.

Ps: Used a "text pdf" and a regular B&W djvu.
```

Original comment by `lanos...@gmail.com` on 4 Oct 2011 at 6:40

**Comment by GoogleCodeExporter** [comment]
```
In 1.2.2-rc4 version this problem arise, see video:
http://www.youtube.com/watch?v=Ex-qCjh2-Hk

In my opinion it seems that this is the same problem as before but now is 
extremely more noticeable(at least one of the causes that produced before a non 
fluid scroll)

Also, besides this now more noticeable issue, the other thing that produce a 
non fluid scroll is that it is like when you scroll "the scrolling 
animation"(just to name it) is dropping frames.
This is the exact sensation that, i can't put into word before, it is like when 
you scroll the scrolling is dropping frames, which visually is a little "jerky".

Ps: This "dropping frames" sensation was not ever seen by me in any other app 
that has something to fling or drag, name it a game, ebook reader, image 
viewer, web browser etc....
Perhaps yes, the tablet could stay long using the CPU to bring to the screen 
whatever thing is shown in the app, but at that moment, when the "image" is 
already on screen(loaded in ram), the fling&drag sensation is almost identical 
in all apps and is very smooth.

thanks
```

Original comment by `lanos...@gmail.com` on 5 Oct 2011 at 10:44

**Comment by GoogleCodeExporter** [comment]
```
This "dropping frames" effect is not noticeable in videos, but is a little 
annoying in real life.
Although i must say that it is a lot better now than it was before.

I don't know nothing about programming such a complex program like ebookdroid.
But perhaps(correct me if i'm saying non-senses), you could use a more "generic 
code/approach" to use in scrolling once the bitmap is already loaded in memory. 
It seems to me, that once the bitmap is in RAM it doesn't matter if the 
document was originally a pdf or jpeg or an html page or whatever, all is 
rendered to the same resolution(tablet resolution generally) so all should 
react similarly.
```

Original comment by `lanos...@gmail.com` on 5 Oct 2011 at 10:59

**Comment by GoogleCodeExporter** [comment]
Original comment by `Alexander.V.Kasatkin@gmail.com` on 6 Oct 2011 at 5:28
- Changed state: **Started**

**Comment by GoogleCodeExporter** [comment]
```
Yes. You right. This "jerkiness" is present on slow scroll. It will be fixed in 
next version.
```

Original comment by `Andrei.K...@gmail.com` on 6 Oct 2011 at 7:41

**Comment by GoogleCodeExporter** [comment]
```
Pls try the 1.2.2-rc5 version.
```

Original comment by `Alexander.V.Kasatkin@gmail.com` on 6 Oct 2011 at 9:07

**Comment by GoogleCodeExporter** [comment]
```
Excellent!
Issue displayed in:
http://www.youtube.com/watch?v=Ex-qCjh2-Hk
IS SOLVED in 1.2.2-rc5 version.

Thanks a lot
```

Original comment by `lanos...@gmail.com` on 6 Oct 2011 at 3:04

**Comment by GoogleCodeExporter** [comment]
Original comment by `Alexander.V.Kasatkin@gmail.com` on 6 Oct 2011 at 3:48
- Changed state: **Fixed**
